### PR TITLE
test: E2E to reveal SRP on error boundary screen

### PR DESCRIPTION
**Pull request by cortisiko**
## **Description**

To catch any regressions in the crash screen’s reveal SRP flow, we should add an end-to-end (e2e) test. Since revealing the Secret Recovery Phrase is the user’s last chance to view and securely save it, this test will check that each step in the flow works smoothly: from entering your password to viewing your SRP.

```
Scenario: A user can reveal their SRP while on the crash screen
Given I encounter the crash screen
When I tap linked text to reveal my SRP
And I enter an invalid password
Then I should see the appropriate error
When I enter the correct password
Then I can see my SRP
```

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/ErrorBoundary/index.js [hunk 1]
```diff
@@ -270,6 +269,14 @@ export const Fallback = (props) => {
           </Text>
         }
       />
+
+      {isTest && (
```
Comment (cortisiko) [inline app/components/Views/ErrorBoundary/index.js:273]: I had trouble interacting with the "Save your Secret Recovery Phrase" link inside the bannerAlert component, so I created a new text component outside of it to display the "Reveal SRP" link. This component is hidden behind the isTest flag, ensuring it only shows up in non-production environments.

File: e2e/pages/Browser/TestDApp.js [hunk 2]
```diff
@@ -40,6 +40,12 @@ class TestDApp {
       TestDappSelectorsWebIDs.APPROVE_ERC_20_TOKENS_BUTTON_ID,
     );
   }
+  get invalidSignature() {
```
Comment (cortisiko) [inline e2e/pages/Browser/TestDApp.js:43]: This is the invalid Signature button on the test dapp. It results in a fatal screen which is perfect for my e2e

File: e2e/pages/ErrorBoundaryView/ErrorBoundaryView.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import Matchers from '../../utils/Matchers';
+import Gestures from '../../utils/Gestures';
+import { ErrorBoundarySelectorsText } from '../../selectors/ErrorBoundary/ErrorBoundaryView.selectors';
+
+class ErrorBoundaryView {
```
Comment (cortisiko) [inline e2e/pages/ErrorBoundaryView/ErrorBoundaryView.js:5]: Error Boundary Page object

File: e2e/pages/Network/NetworkApprovalBottomSheet.js [hunk 4]
```diff
@@ -4,14 +4,21 @@ import Gestures from '../../utils/Gestures';
 
 class NetworkApprovalBottomSheet {
   get container() {
-    return Matchers.getElementByID(NetworkApprovalBottomSheetSelectorsIDs.CONTAINER);
+    return Matchers.getElementByID(
```
Comment (cortisiko) [inline e2e/pages/Network/NetworkApprovalBottomSheet.js:7]: Adding Methods to network approval bottom sheet

**Review by chrisleewilcox — DISMISSED** [review]
LGTM

**Review by EtherWizard33 — APPROVED** [review]
LGTM

**Review by chrisleewilcox — APPROVED** [review]
LGTM

**Pull request merged**